Marlin/Marlin_main.cpp

```diff
@@ -10257,14 +10290,23 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
       UNUSED(fr_mm_s);
       UNUSED(no_move);
 
-      // Set the new active extruder
-      active_extruder = tmp_extruder;
+      #if ENABLED(SWITCHING_EXTRUDER) && !DONT_SWITCH
+
+        stepper.synchronize();
+        move_extruder_servo(tmp_extruder);
+
+      #elif ENABLED(MK2_MULTIPLEXER)
+
+        if (tmp_extruder >= E_STEPPERS)
+          return invalid_extruder_error(tmp_extruder);
+
+        select_multiplexed_stepper(tmp_extruder);
```

Comment:
>We should probably also add a sanity check to ensure that MIXING_EXTRUDER and MK2_MULTIPLEXER are not both enabled.

Thinking about this some more, `MIXING_EXTRUDER` probably would work with `MK2_MULTIPLEXER` so long as `select_multiplexed_stepper` is a relatively quick operation since the steps are actually sent out to each stepper one at a time in `Stepper`.